app/api/utils.py

```diff
@@ -423,6 +423,38 @@ def parse(self, file):
             yield data
 
 
+class FasttextParser(FileParser):
+    """
+    Upload files in fasttext format.
+    Labels are marked with the __label__ key 
+    and the corresponding text comes afterwards in the same row
+    For example:
+    ```
+    __label__dog poodle
+    __label__house mansion 
+    ```
+    """
+    def parse(self, file):
+        file = EncodedIO(file)
+        file = io.TextIOWrapper(file, encoding=file.encoding)
+        data = []
+        for _, line in enumerate(file, start=0):
+            if len(data) >= settings.IMPORT_BATCH_SIZE:
+                yield data
+                data = []
+            j = {'text': '', 'labels' :[]} 
+            for word in line.split(' '):
+                if '__label__' in word:
+                    j['labels'].append(word.split('__label__')[1])
```

Comment:
`word[len('__label__'):]`

Reply:
check if there is a "label text" that comes after the marker. otherwise exception.